app/controllers/sources_controller.rb

```diff
@@ -94,4 +93,15 @@ def source_params
   def load_source_set
     @source_set = SourceSet.friendly.find(params[:source_set_id])
   end
+
+  def parse_dpla_item(dpla_item)
+    return unless dpla_item.present?
+    @digital_resource_url = dpla_item.try(:url)
+    @provider_name = dpla_item.try(:provider).try(:name)
+    data_provider = dpla_item.try(:source)
+    intermediate_provider = dpla_item.try(:intermediate_provider)
+    @contributing_institution = 
+      [data_provider, intermediate_provider].compact.join('; ')
+    @title = Array(dpla_item.try(:title)).flatten.first
+  end
```

Comment:
This is getting to be a lot of logic for a controller.  I'm curious to know if you think it would be beneficial to instantiate `dpla_item` as an object, with methods to get the `digital_resource_url`, `provider_name`, etc.?  I am on the fence about it myself.  I don't think we will need to parse any more attributes than what you see here, at least in the foreseeable future, so I don't anticipate this `parse_dpla_item` method to get any bigger.